Dockerfile

```diff
@@ -6,20 +6,53 @@ RUN mkdir -p /tmp && chown -R www-data:www-data /tmp && chmod -R 777 /tmp
 RUN mkdir -p /var/run/ && chown -R www-data:www-data /var/run && chmod -R 777 /var/run
 USER www-data
 
-FROM base AS prod
+# We install the dependencies in a separate layer as the frontend image also needs them
+FROM base AS vendor
 USER root
-COPY . /var/www/html
+
+# Copying the full context is not great for caching
+# But composer install executes symfony commands that need the full context
+COPY . /var/www/html/
 
 RUN composer install
+
+FROM node:22 AS frontend
```

Comment:
I would use an ARG for the node version

Reply:
done

Reply:
In the end I'm not sure if this is a good idea.

having `FROM node:22` ensures that renovate can parse it and propose updates
Using `FROM node:${NODE_VERSION}` is more explicit for humans but will not help renovate

If we use the arg, then it would be good to also enable this feature from Renovate; https://docs.renovatebot.com/modules/manager/regex/#advanced-capture
In essence it adds a custom regex to parse the dockerfile and still detect the NODE_VERSION arg

Reply:
On typesense I had no issue with renovate and args. I think it just works.

> Also, Renovate will automatically expand variables and [ARG directives](https://docs.docker.com/reference/dockerfile/#understand-how-arg-and-from-interact):

https://docs.renovatebot.com/modules/manager/dockerfile/

Reply:
okay let's keep the arg then